### PR TITLE
Add matrix to runs-on in snaps

### PR DESCRIPTION
**Pull request by gabrielcocenza**
This way it's possible to build and release snaps using arm64 for the projects.

Added arm64 build and release for the dcgm snap

### CODE REVIEW

**Review by aieri — APPROVED** [review]

File: terraform-plans/configs/dcgm-snap_main.tfvars [hunk 1]
```diff
@@ -10,7 +10,9 @@ templates = {
   check = {
     source      = "./templates/github/snap_check.yaml.tftpl"
     destination = ".github/workflows/check.yaml"
-    vars        = {}
+    vars        = {
+      runs_on = "[[ubuntu-22.04], [Ubuntu_ARM64_4C_16G_01]]",
```
Comment (samuelallan72) [inline terraform-plans/configs/dcgm-snap_main.tfvars:14]: It's hard to see if this works without merging; could you test this by opening a draft PR to dcgm-snap with the changes to verify that the workflow passes? :)

**Review by samuelallan72 — CHANGES_REQUESTED** [review]
This looks great thanks, but it would be good to see this tested on dcgm-snap before it's merged here. :)

**Comment by Pjack** [comment]
@gabrielcocenza Why we need to support ARM64 for COU project ?

**Comment by gabrielcocenza** [comment]
> Why we need to support ARM64 for COU project ?

@Pjack  We don't. I'm just adding arm64 for `DCGM`. The other projects I'm adding a variable to be populated with what to run. In case of COU, tailscale and etc it's going to run just on amd64

**Comment by Pjack** [comment]
> > Why we need to support ARM64 for COU project ?
> 
> @Pjack We don't. I'm just adding arm64 for `DCGM`. The other projects I'm adding a variable to be populated with what to run. In case of COU, tailscale and etc it's going to run just on amd64

Understand. 

This is another example of why we need to implement hierarchical variables. Ideally, modifying `charmed-openstack-upgrader_main.tfvars` should be unnecessary when we only want to adjust the behavior in `dcgm-snap_main.tfvars`

File: terraform-plans/configs/dcgm-snap_main.tfvars [hunk 2]
```diff
@@ -20,7 +22,9 @@ templates = {
   release = {
     source      = "./templates/github/snap_release.yaml.tftpl"
     destination = ".github/workflows/release.yaml"
-    vars        = {}
+    vars        = {
+       runs_on = "[[ubuntu-22.04], [Ubuntu_ARM64_4C_16G_01]]",
```
Comment (chanchiwai-ray) [inline terraform-plans/configs/dcgm-snap_main.tfvars:26]: `Ubuntu_ARM64_4C_16G_02` ?

File: terraform-plans/configs/dcgm-snap_main.tfvars (hunk 1 above)
Comment (gabrielcocenza) [reply terraform-plans/configs/dcgm-snap_main.tfvars]: That was a good suggestion. I had no idea how many things I had to fix until is finally working :grimacing: 

Take a look at https://github.com/canonical/dcgm-snap/pull/15

File: terraform-plans/configs/dcgm-snap_main.tfvars (hunk 2 above)
Comment (gabrielcocenza) [reply terraform-plans/configs/dcgm-snap_main.tfvars]: I've changed to use the self-hosted while the IS give us access to the arm runner. But yes, that should be changed to `Ubuntu_ARM64_4C_16G_02` later.

@chanchiwai-ray do you need to start running arm64 on smartctl also? If you need I can include at this PR

File: terraform-plans/configs/dcgm-snap_main.tfvars (hunk 1 above)
Comment (samuelallan72) [reply terraform-plans/configs/dcgm-snap_main.tfvars]: Wow!  Nice work :grin: I vote we merge https://github.com/canonical/dcgm-snap/pull/15 now that it's passing CI; then when we merge this, we can verify that it's produced the same result easily. :)

**Review by samuelallan72 — APPROVED** [review]

**Review by jneo8 — APPROVED** [review]

File: terraform-plans/configs/dcgm-snap_main.tfvars (hunk 1 above)
Comment (gabrielcocenza) [reply terraform-plans/configs/dcgm-snap_main.tfvars]: Now that is merged, I'm solving this thread. Thanks Sam!

File: terraform-plans/configs/dcgm-snap_main.tfvars (hunk 2 above)
Comment (gabrielcocenza) [reply terraform-plans/configs/dcgm-snap_main.tfvars]: Solving this thread once is using now the self-hosted. We can change once IS gives permission to use the GitHub runners and you can open a new PR when you want to add arm64 for smartctl.

**Pull request merged**